scio-test/src/main/scala/com/spotify/scio/testing/CoderAssertions.scala

```diff
@@ -30,102 +32,175 @@ import scala.reflect.ClassTag
 object CoderAssertions {
   private lazy val DefaultPipelineOptions = PipelineOptionsFactory.create()
 
-  implicit class CoderAssertionsImplicits[T](private val value: T) extends AnyVal {
+  type CoderAssertionT[T] = AssertionContext[T] => Assertion
+  type CoderAssertion = AssertionContextBase => Assertion
```

Comment:
I'm wondering if this might be overloading the name `CoderAssertion` too much... maybe we could rename this something like `GenericCoderAssertion`? Although that might be confusing too if people hear "Generic" and think we're talking about Avro 😓 wdyt?

Reply:
Hmm.. What about `CoderAssertionBase`, like I did with `AssertionContextBase`?